core/src/test/java/com/datastax/oss/driver/internal/core/metadata/MetadataManagerTest.java

```diff
@@ -290,6 +287,8 @@ public void should_remove_node() {
   private static class TestMetadataManager extends MetadataManager {
 
     private List<MetadataRefresh> refreshes = new CopyOnWriteArrayList<>();
+    private int addNodeCount = 0;
```

Comment:
These 2 fields probably should be volatile.

Reply:
Good catch. I did try to verify that the tests run in parallel don't trample each other here, and it seemed like they didn't, but `volatile` here makes sense.